test/e2e/unmanaged_dns_test.go

```diff
@@ -240,8 +263,35 @@ func TestUnmanagedDNSToManagedDNSInternalIngressController(t *testing.T) {
 		t.Fatalf("failed to update ingresscontroller %s: %v", name, err)
 	}
 
```

Comment:
Should this step wait for stability too?
```suggestion
// Wait for the load balancer and DNS to reach stable conditions.
	if err := waitForIngressControllerCondition(t, kclient, 10*time.Minute, name, append(availableConditionsForIngressControllerWithLoadBalancer, operatorProgressingFalse)...); err != nil {
		t.Fatalf("failed to observe expected conditions: %v", err)
	}
```

Reply:
We do wait for DNS stable conditions on [line 300](https://github.com/openshift/cluster-ingress-operator/pull/845/files/bcf092c41b7cd5d1e4b74197fe2d45cbda038d6e#diff-a660dcb9b4adcc19e854b3f294ec260efabed12bb55eb2f2b8fb03ff6eb92527R300) before we do our verification test.

Reply:
I've seen (and written) e2e tests where whenever there's an operation that has side effects, you call `waitForIngressControllerCondition`  Here, since he called `updateIngressControllerSpecWithRetryOnConflict`, it might be beneficial to call `waitForIngressControllerCondition`.

Reply:
I agree, but we do exactly `waitForIngressControllerCondition` on [line 300](https://github.com/openshift/cluster-ingress-operator/pull/845/files/bcf092c41b7cd5d1e4b74197fe2d45cbda038d6e#diff-a660dcb9b4adcc19e854b3f294ec260efabed12bb55eb2f2b8fb03ff6eb92527R300) after this update operation.

It's not immediately below the IC update because we can go do some other things (like check for lbService update) before we `waitForIngressControllerCondition`, being more efficient, but it still comes after the update like you suggest to make sure we don't have any side effects.

Is that sufficient?

Reply:
It's a toss up on whether you wait for things to come back to normal after a change, or you wait and redo things over and over in a loop, adding time when you find you need to tweak the test.